packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/index.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import { includes } from 'ramda'
+
+import Currencies from '@core/exchange/currencies'
+import { CoinType, FiatType } from '@core/types'
+import DataError from 'components/DataError'
+import { actions } from 'data'
+import { Analytics, EarnStepMetaData } from 'data/types'
+import { useRemote } from 'hooks'
+
+import Loading from '../Staking.template.loading'
+import { getData } from './AccountSummary.selectors'
+import AccountSummary from './AccountSummary.template.success'
+import Unsupported from './AccountSummary.template.unsupported'
+
+const AccountSummaryContainer = (props: OwnProps) => {
+  const { coin, handleClose, showSupply, stepMetadata, walletCurrency } = props
+  const { data, error, isLoading, isNotAsked } = useRemote(getData)
+  const dispatch = useDispatch()
+  const unsupportedCurrencies = [Currencies.TWD.code, Currencies.CLP.code]
+  const isFiatCurrencySupported = !includes(walletCurrency, unsupportedCurrencies)
+
+  useEffect(() => {
+    dispatch(actions.components.interest.fetchStakingLimits())
+  }, [])
+
+  useEffect(() => {
+    if (isFiatCurrencySupported) {
+      dispatch(
+        actions.analytics.trackEvent({
+          key: Analytics.WALLET_STAKING_DETAIL_VIEWED,
+          properties: {
+            currency: coin
+          }
+        })
+      )
+    }
+  }, [coin])
+
+  const handleDepositClick = () => {
```

Comment:
You could technically get a little performance boost if you turn these into a hook like `useCallback(() => {...}, [coin])`